src/ssa/local_ssa.cpp

```diff
@@ -537,37 +537,58 @@ void local_SSAt::build_guard(locationt loc)
   (--nodes.end())->equalities.push_back(equality);
 }
 
-/// turns assertions into constraints
-void local_SSAt::build_assertions(locationt loc)
+/// Recursively go through expr and look for an equality with a pointer
+/// object on the right side and construct a disjunct of possible
+/// concrete assignments.
+bool local_SSAt::get_deallocated_disjunction(const exprt &expr, exprt &result)
 {
-  if(loc->is_assert())
+  if(expr.id()==ID_equal && expr.op1().id()==ID_pointer_object &&
+     expr.op1().op0().id()==ID_symbol)
```

Comment:
Let's do this properly:
```suggestion
  if(expr.id()==ID_equal)
  {
    auto &rhs=to_equal_exprt(expr).rhs();
    if(rhs.id()==ID_pointer_object && rhs.op0().id()==ID_symbol)
    {
      const symbol_exprt &dealloc_symbol=to_symbol_exprt(rhs.op0());
      std::string id=id2string(dealloc_symbol.get_identifier());
      if (id.find("__CPROVER_deallocated")!=std::string::npos)
      {
        exprt::operandst d;
        ....
```